src/main/java/org/dependencytrack/util/AnalysisCommentUtil.java

```diff
@@ -24,53 +24,43 @@
-            analysisStateChange = true;
-            qm.makeAnalysisComment(analysis, String.format("Analysis: %s → %s", analysis.getAnalysisState(), analysisState), commenter);
-        }
-        return analysisStateChange;
+        return makeCommentIfChanged("Analysis", qm, analysis, Objects.requireNonNullElse(analysis.getAnalysisState(), AnalysisState.NOT_SET), analysisState, commenter);
     }
 
     public static void makeJustificationComment(final QueryManager qm, final Analysis analysis, final AnalysisJustification analysisJustification, final String commenter) {
-        if (analysisJustification != null) {
-            if (analysis.getAnalysisJustification() == null && AnalysisJustification.NOT_SET != analysisJustification) {
-                qm.makeAnalysisComment(analysis, String.format("Justification: %s → %s", AnalysisJustification.NOT_SET, analysisJustification), commenter);
-            } else if (analysis.getAnalysisJustification() != null && analysisJustification != analysis.getAnalysisJustification()) {
-                qm.makeAnalysisComment(analysis, String.format("Justification: %s → %s", analysis.getAnalysisJustification(), analysisJustification), commenter);
-            }
-        }
+        makeCommentIfChanged("Justification", qm, analysis, Objects.requireNonNullElse(analysis.getAnalysisJustification(), AnalysisJustification.NOT_SET), analysisJustification, commenter);
     }
 
     public static void makeAnalysisResponseComment(final QueryManager qm, final Analysis analysis, final AnalysisResponse analysisResponse, final String commenter) {
-        if (analysisResponse != null) {
-            if (analysis.getAnalysisResponse() == null && analysis.getAnalysisResponse() != analysisResponse) {
-                qm.makeAnalysisComment(analysis, String.format("Vendor Response: %s → %s", AnalysisResponse.NOT_SET, analysisResponse), commenter);
-            } else if (analysis.getAnalysisResponse() != null && analysis.getAnalysisResponse() != analysisResponse) {
-                qm.makeAnalysisComment(analysis, String.format("Vendor Response: %s → %s", analysis.getAnalysisResponse(), analysisResponse), commenter);
-            }
-        }
+        makeCommentIfChanged("Vendor Response", qm, analysis, Objects.requireNonNullElse(analysis.getAnalysisResponse(), AnalysisResponse.NOT_SET), analysisResponse, commenter);
     }
 
     public static void makeAnalysisDetailsComment(final QueryManager qm, final Analysis analysis, final String analysisDetails, final String commenter) {
-        if (analysisDetails != null && !analysisDetails.equals(analysis.getAnalysisDetails())) {
-            final String message = "Details: " + analysisDetails.trim();
-            qm.makeAnalysisComment(analysis, message, commenter);
-        }
+        makeCommentIfChanged("Details", qm, analysis, Objects.requireNonNullElse(analysis.getAnalysisDetails(), "None"), analysisDetails, commenter);
```

Comment:
Note how details comments did not follow the `old -> new` format, but instead simply logged the new value. Reason being that details can be arbitrarily long, such that including both old and new value in the same comment would flood the audit trail.

The previous value will already be present in the audit trail, so it is not lost.

Reply:
I understand that this would introduce "noise". However, I am personally more leaning into logging every change _explicitly_. This is debatable. I understand that. But I think this is more like an _output_ problem, not an audit problem.

Reply:
Hmmm good point. We should probably add some sort of separator between trail entries in the UI, to make it more obvious where one entry ends and the next begins. Do you think that would be sufficient to address your concern?

Reply:
That was exactly my idea to be honest - if HTML is escaped properly, adding a simple `<hr>` tag between entries should be enough.